pkg/gpu/nvidia/mig/mig.go

```diff
@@ -257,3 +266,19 @@ func (d *DeviceManager) discoverNumGPUs() (int, error) {
 	}
 	return numGPUs, nil
 }
+
+func (d *DeviceManager) topology(deviceIndex string) (*pluginapi.TopologyInfo, error) {
+	index, err := strconv.Atoi(deviceIndex)
+	if err != nil {
+		return nil, fmt.Errorf("unable to convert deviceIndex %q string to int: %v", deviceIndex, err)
+	}
+
+	if nvmlutil.NvmlDeviceInfo == nil {
+		nvmlutil.NvmlDeviceInfo = &nvmlutil.DeviceInfo{}
+	}
+	device, ret := nvmlutil.NvmlDeviceInfo.DeviceHandleByIndex(index)
```

Comment:
In nvmlutil.go, you have function `MigDeviceHandleByIndex` and `MigMode`, are they needed?

Reply:
removed those methods.

Reply:
Here you are getting the NUMA info for the whole GPU, and put it to all partitions in this GPU, does it mean all partitions in the same GPU share the same NUMA info?

Reply:
Correct! all the partitions will have same numa info as the parent.